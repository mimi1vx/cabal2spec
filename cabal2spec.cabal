name:                   cabal2spec
version:                2.0.0
synopsis:               Convert Cabal files into rpm spec files
homepage:               https://github.com/peti/cabal2spec
license:                GPL-3
license-file:           LICENSE
author:                 Peter Simons, Bryan O'Sullivan, Jens Petersen
maintainer:             simons@cryp.to
category:               Distribution
build-type:             Simple
cabal-version:          >=1.10
tested-with:            GHC == 7.4.2, GHC == 7.6.3, GHC == 7.8.4, GHC == 7.10.3
                      , GHC == 8.0.2, GHC == 8.2.2

description:
  Convert Cabal files into a spec file suitable for building the package with
  the RPM package manager. This tools primarily targets the SUSE/openSUSE
  familiy of distributions. See <http://hackage.haskell.org/package/cabal-rpm>
  for a different project that has a more ambitious scope, i.e. it's trying to
  support other RPM-based distributions, too.

source-repository head
  type:                 git
  location:             git://github.com/peti/cabal2spec.git

library
  hs-source-dirs:       src
  ghc-options:          -Wall
  build-depends:        base < 5
                      , Cabal > 2
                      , filepath
                      , time >= 1.5
  exposed-modules:      Cabal2Spec
  default-language:     Haskell2010

executable cabal2spec
  main-is:              Main.hs
  hs-source-dirs:       cabal2spec
  default-language:     Haskell2010
  ghc-options:          -Wall
  build-depends:        base < 5
                      , Cabal > 2
                      , cabal2spec
                      , filepath

test-suite regression-test
  type:                 exitcode-stdio-1.0
  main-is:              Main.hs
  hs-source-dirs:       test
  build-depends:        base
                      , Cabal > 2
                      , cabal2spec
                      , filepath
                      , tasty
                      , tasty-golden
  ghc-options:          -Wall
  default-language:     Haskell2010
