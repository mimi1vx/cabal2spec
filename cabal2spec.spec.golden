#
# spec file for package cabal2spec
#
# Copyright (c) 2018 SUSE LINUX GmbH, Nuernberg, Germany.
#
# All modifications and additions to the file contributed by third parties
# remain the property of their copyright owners, unless otherwise agreed
# upon. The license for this file, and modifications and additions to the
# file, is the same license as for the pristine package itself (unless the
# license for the pristine package is not an Open Source License, in which
# case the license is the MIT License). An "Open Source License" is a
# license that conforms to the Open Source Definition (Version 1.9)
# published by the Open Source Initiative.

# Please submit bugfixes or comments via http://bugs.opensuse.org/
#

%global pkg_name cabal2spec

%bcond_with tests

Name:           %{pkg_name}
Version:        2.0.0
Release:        0
Summary:        Convert Cabal files into rpm spec files
Group:          Development/Libraries/Haskell

License:        GPL-3.0+
Url:            https://hackage.haskell.org/package/%{name}
Source0:        https://hackage.haskell.org/package/%{name}-%{version}/%{name}-%{version}.tar.gz

BuildRequires:  ghc-Cabal-devel
BuildRequires:  ghc-rpm-macros
BuildRequires:  chrpath
BuildRequires:  ghc-directory-devel
BuildRequires:  ghc-filepath-devel
BuildRequires:  ghc-time-devel
%if %{with tests}
BuildRequires:  ghc-tasty-devel
BuildRequires:  ghc-tasty-golden-devel
%endif

%description
Convert Cabal files into a spec file suitable for building the package with the
RPM package manager. This tools primarily targets the SUSE/openSUSE familiy of
distributions. See <http://hackage.haskell.org/package/cabal-rpm> for a
different project that has a more ambitious scope, i.e. it's trying to support
other RPM-based distributions, too.


%package -n ghc-%{name}
Summary:        Haskell %{name} library
Group:          System/Libraries

%description -n ghc-%{name}
This package provides the Haskell %{name} shared library.


%package -n ghc-%{name}-devel
Summary:        Haskell %{name} library development files
Group:          Development/Libraries/Haskell
Requires:       ghc-compiler = %{ghc_version}
Requires(post): ghc-compiler = %{ghc_version}
Requires(postun): ghc-compiler = %{ghc_version}
Requires:       ghc-%{name} = %{version}-%{release}

%description -n ghc-%{name}-devel
This package provides the Haskell %{name} library development files.


%prep
%setup -q

%build
%ghc_lib_build

%install
%ghc_lib_install
%ghc_fix_rpath %{pkg_name}-%{version}

%check
%cabal_test

%post -n ghc-%{name}-devel
%ghc_pkg_recache

%postun -n ghc-%{name}-devel
%ghc_pkg_recache

%files
%doc LICENSE
%{_bindir}/%{name}

%files -n ghc-%{name} -f ghc-%{name}.files
%defattr(-,root,root,-)
%doc LICENSE

%files -n ghc-%{name}-devel -f ghc-%{name}-devel.files
%defattr(-,root,root,-)

%changelog
